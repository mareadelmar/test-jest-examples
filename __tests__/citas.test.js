import Citas from "../js/classes/Citas";

// Probar los métodos de agregar, editar y eliminar citas

describe("clase Citas", () => {
	const citas = new Citas();

	const id = Date.now(); // lo dejamos acá para que sea global y poder usarlo para probar el método editar y eliminar.

	test("Agregar una nueva cita", () => {
		const citaObj = {
			id,
			mascota: "Pepe",
			propietario: "Juan",
			telefono: "123456",
			fecha: "10-12",
			hora: "11:00",
			sintomas: "Habla",
		};

		citas.agregarCita(citaObj);

		expect(citas).toMatchSnapshot();
	});

	test("Editar una cita", () => {
		// cambiamos algunos datos para verlo reflejado en el snapshot y saber que está bien implementado

		const citaActualizada = {
			id,
			mascota: "Pepe Loco",
			propietario: "Juan",
			telefono: "123456",
			fecha: "10-12",
			hora: "11:00",
			sintomas: "Habla sin parar",
		};

		citas.editarCita(citaActualizada);

		expect(citas).toMatchSnapshot(); // la próxima ejecución requiere actualizar el snapshot con -- -u
	});

	test("Eliminar cita", () => {
		citas.eliminarCita(id);

		expect(citas).toMatchSnapshot();
	});
});
