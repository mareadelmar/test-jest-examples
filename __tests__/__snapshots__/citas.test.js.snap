// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clase Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12",
      "hora": "11:00",
      "id": 1640006577218,
      "mascota": "Pepe",
      "propietario": "Juan",
      "sintomas": "Habla",
      "telefono": "123456",
    },
  ],
}
`;

exports[`clase Citas Editar una cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12",
      "hora": "11:00",
      "id": 1640006577218,
      "mascota": "Pepe Loco",
      "propietario": "Juan",
      "sintomas": "Habla sin parar",
      "telefono": "123456",
    },
  ],
}
`;

exports[`clase Citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
